src/managers/input/Touch.cpp

```diff
@@ -32,12 +32,12 @@ void CInputManager::onTouchDown(wlr_touch_down_event* e) {
 
     if (m_sTouchData.touchFocusWindow) {
         if (m_sTouchData.touchFocusWindow->m_bIsX11) {
-            local = g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchFocusWindow->m_vRealPosition.goalv();
+            local = (g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchFocusWindow->m_vRealPosition.goalv()) * m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy;
+            m_sTouchData.touchSurfaceOrigin = (g_pInputManager->getMouseCoordsInternal() * m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy) - local;
```

Comment:
(coords - windowpos) * scale

Reply:
I'll try to explain my changes and how I think they relate to the formula above. But all my changes revolve around : add a scaling factor where needed in already existing code / computation formulas.

The original code is (if the window is X11) :

```
local = g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchFocusWindow->m_vRealPosition.goalv();
```

I understand `g_pInputManager->getMouseCoordsInternal()` as `coords`.
I understand `m_sTouchData.touchFocusWindow->m_vRealPosition.goalv()` as `windowpos`.

I scaled this whole expression by `m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy` which I understand as `scale`.

The result is : 

```
local = (g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchFocusWindow->m_vRealPosition.goalv()) * m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy;
```

Which corresponds to `(coords - windowpos) * scale` in my understanding. After several tests is works for the initial touch down, which is what this function is all about. 

However this function also prepares another, for touch moves, and stores a `touchSurfaceOrigin`. 

The original code uses two distincts paths if the window is X11 or not to compute the `local` variable. In the X11 case this variable units are X11 scaled units, in the other case the units are global screen scale units. What is important is that they differ.

The original code modify this variable this way : 

```
m_sTouchData.touchSurfaceOrigin = g_pInputManager->getMouseCoordsInternal() - local;
```

As `getMouseCoordsInternal()` is in global screen scale units and local may be in one or the other unit, I was forced to compute `touchSurfaceOrigin` differently in the two cases.

The modified code in the case if an X11 window becomes : 

```
m_sTouchData.touchSurfaceOrigin = (g_pInputManager->getMouseCoordsInternal() * m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy) - local;
```

`local` is already in X11 units, I have to change the coords again accordingly. (or use a variable and scale it once and for all, but I chose to modify the original code as few as possible). 

All this code prepares what will happen next in `onTouchMove()` that I describe in the other conversation.

Reply:
you absolutely cannot scale the getMouseCoordsInternal by any scale. Those are logical wayland coordinates.

Sorry, I also forgot what touchSurfaceOrigin was.

Fixed formula:
`m_sTouchData.touchSurfaceOrigin = m_sTouchData.touchFocusWindow->m_vRealPosition.goalv();`
using the fact that X11 windows are single-surface.

Reply:
Well, I did not mean scale them, I mean doing a computation to output the correct event value. 

Ok, simpler way to compute origin.